### PR TITLE
Tested beans_remove_output

### PR DESCRIPTION
**Pull request by hellofromtonya**
1. Added integration and unit tests.
2. Improved function's documentation.
3. Added a crazy high priority number to ensure it fires last.

### CODE REVIEW

File: tests/phpunit/integration/api/html/beansRemoveOutput.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Class Tests_BeansRemoveOutput
+ *
+ * @package Beans\Framework\Tests\Integration\API\HTML
+ * @group   api
+ * @group   api-html
+ */
+class Tests_BeansRemoveOutput extends HTML_Test_Case {
+
+	/**
+	 * Test beans_remove_output() should return an _Beans_Anonymous_Filters instance.
```
Comment (christophherr) [inline tests/phpunit/integration/api/html/beansRemoveOutput.php]: `return a _Beans_Anonymous_Filters instance.`
(or is the underscore pronounced?)

File: tests/phpunit/integration/api/html/beansRemoveOutput.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	 */
+	public function test_should_return_anonymous_filter_instance() {
+		$this->assertInstanceOf( _Beans_Anonymous_Filters::class, beans_remove_output( 'beans_post_meta_item_date' ) );
+	}
+
+	/**
+	 * Test beans_remove_output() should register callback to the "{$id}_output" filter hook.
+	 */
+	public function test_should_return_register_callback_to_id_output() {
```
Comment (christophherr) [inline tests/phpunit/integration/api/html/beansRemoveOutput.php]: maybe add `_filter` at the end? 
(`id_output_filter`)

File: tests/phpunit/unit/api/html/beansRemoveOutput.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			Monkey\Functions\expect( 'beans_add_filter' )
+				->once()
+				->with( "{$id}_output", false, 99999999 )
+				->andReturn( false );
+
+			beans_remove_output( $id );
+		}
+
+		// Tests are focused above on ensuring beans_apply_filters() is called with the right arguments.
+		$this->assertTrue( true );
```
Comment (christophherr) [inline tests/phpunit/unit/api/html/beansRemoveOutput.php]: Are the two lines above (L#66-L#67) leftovers from a different test?

Comment (hellofromtonya) [reply tests/phpunit/unit/api/html/beansRemoveOutput.php]: No, there are needed as PHPUnit requires an assert.  The comment lets you know that we are testing with Brain Monkey's expect. BUT I had the wrong function listed as it should be `beans_add_filter` and not `beans_apply_filters`.

**Pull request merged**